python_modules/dagster/dagster/core/execution/plan/plan.py

```diff
@@ -758,19 +758,32 @@ def build_subset_plan(
         step_output_versions = check.opt_dict_param(
             step_output_versions, "step_output_versions", key_type=StepOutputHandle, value_type=str
         )
-        step_handles_to_execute = [StepHandle.parse_from_key(key) for key in step_keys_to_execute]
-
+        step_handles_to_execute_set = set()
         bad_keys = []
-        for handle in step_handles_to_execute:
+
+        for key in step_keys_to_execute:
+            handle = StepHandle.parse_from_key(key)
+
             if handle not in self.step_dict:
+                if isinstance(handle, ResolvedFromDynamicStepHandle):
+                    unresolved_handle = UnresolvedStepHandle(solid_handle=handle.solid_handle)
+                    if unresolved_handle in self.step_dict: # ok if the unresolved version is present
+                        step_handles_to_execute_set.add(unresolved_handle)
+                        continue
```

Comment:
add some comments to clarify the extrapolation happening here

Reply:
After further investigation, i found the root cause isn't about the parsing or unresolved vs resolved. the root cause is that `self.step_dict` here doesn't have a bunch of `ResolvedFromDynamicStepHandle` that is selected here - and this is because in [1] we explicitly dropped the steps we intent to re-execute. So this may have not been a bug but a feature and I think we should loosen that constraints (i.e. not drop these steps when generating dynamic_mappings from parent_state) and allow [2]